packages/perspective-viewer-d3fc/src/js/legend/styling/cropCellContents.js

```diff
@@ -0,0 +1,41 @@
+/******************************************************************************
+ *
+ * Copyright (c) 2017, the Perspective Authors.
+ *
+ * This file is part of the Perspective library, distributed under the terms of
+ * the Apache License 2.0.  The full license can be found in the LICENSE file.
+ *
+ */
+
+import * as d3 from "d3";
+import {isElementOverflowing} from "../../utils/utils";
+
+export function cropCellContents(legendDiv) {
+    const legendCells = legendDiv.select("g.legendCells");
+    const legendDivRect = legendDiv.node().getBoundingClientRect();
+
+    if (!isElementOverflowing(legendDivRect, legendCells.node().getBoundingClientRect())) {
+        return;
+    }
+
+    const overflowingCells = legendCells
+        .selectAll(".label")
+        .nodes()
+        .filter(cell => isElementOverflowing(legendDivRect, cell.getBoundingClientRect()));
+
+    const svg = legendDiv.select(".legend");
+
+    overflowingCells.forEach(cell => {
```

Comment:
Are we essentially looping over the overflowing cells twice? Would it be better to loop over all the cells and do the `isElementOverflowing` check in the `forEach`?

Reply:
Also, can this be more simply done using D3's `each()`?

```javascript
const overflowingCells = legendCells
        .selectAll(".label")
        .each((d, i, nodes) => {
          const cell = nodes[i];
          if (isElementOverflowing(/*...*/) {
            // ...
        });
```

Reply:
Or is the data `d` the actual label anyway? In which case:

```javascript
const overflowingCells = legendCells
        .selectAll(".label")
        .text((d, i, nodes) => {
          const cell = nodes[i];
          if (isElementOverflowing(/*...*/) {
            // ...
            return d.substring(0, cutoffCharIndex - 3);
          } else {
            return d;
          }
        });
```

Reply:
Accepted the change